plugins/Wallet/js/transactions.js

```diff
@@ -92,13 +92,13 @@ $('#transaction-page').on('input', updateTransactionPage);
 function checkCriteria(txn) {
 	// Checks each criteria value if it's relaxed or if the transaction passes
 	// Failing one check returns false but passing all checks returns true
-	return (criteria.endTime   === undefined || txn.confirmationtimestamp <= criteria.endTime)   &&
-	       (criteria.startTime === undefined || txn.confirmationtimestamp >= criteria.startTime) &&
-	       (criteria.maxValue  === undefined || txn.value                 <= criteria.maxValue)  &&
-	       (criteria.minValue  === undefined || txn.value                 >= criteria.minValue)  &&
-	       (criteria.currency  === undefined || criteria.currency.indexOf(txn.currency) !== -1)  &&
-	       (criteria.inputs    === true      || txn.value                 >= 0)                  &&
-	       (criteria.outputs   === true      || txn.value                 <= 0);
+	return (criteria.endTime   === null || txn.confirmationtimestamp <= criteria.endTime)   &&
+	       (criteria.startTime === null || txn.confirmationtimestamp >= criteria.startTime) &&
+	       (criteria.maxValue  === null || txn.value                 <= criteria.maxValue)  &&
+	       (criteria.minValue  === null || txn.value                 >= criteria.minValue)  &&
+	       (criteria.currency  === null || criteria.currency.indexOf(txn.currency) !== -1)  &&
```

Comment:
what's the advantage of switching to `null` here? I assume it protects against using non-existent fields, e.g. `criteria.foo`?